src/sentry/static/sentry/app/utils/discover/fields.tsx

```diff
@@ -261,83 +261,154 @@ export type Aggregation = {
-  'device.uuid': 'string',
-  'device.arch': 'string',
-  'device.battery_level': 'number',
-  'device.orientation': 'string',
-  'device.simulator': 'boolean',
-  'device.online': 'boolean',
-  'device.charging': 'boolean',
-  'geo.country_code': 'string',
-  'geo.region': 'string',
-  'geo.city': 'string',
-  'error.type': 'string',
-  'error.value': 'string',
-  'error.mechanism': 'string',
-  'error.handled': 'boolean',
-  'stack.abs_path': 'string',
-  'stack.filename': 'string',
-  'stack.package': 'string',
-  'stack.module': 'string',
-  'stack.function': 'string',
-  'stack.in_app': 'boolean',
-  'stack.colno': 'number',
-  'stack.lineno': 'number',
-  'stack.stack_level': 'number',
+  [FieldKey.TRANSACTION]: 'string',
+  [FieldKey.USER]: 'string',
+  [FieldKey.USER_ID]: 'string',
+  [FieldKey.USER_EMAIL]: 'string',
+  [FieldKey.USER_USERNAME]: 'string',
+  [FieldKey.USER_IP]: 'string',
+  [FieldKey.SDK_NAME]: 'string',
+  [FieldKey.SDK_VERSION]: 'string',
+  [FieldKey.HTTP_METHOD]: 'string',
+  [FieldKey.HTTP_URL]: 'string',
+  [FieldKey.OS_BUILD]: 'string',
+  [FieldKey.OS_KERNEL_VERSION]: 'string',
+  [FieldKey.DEVICE_NAME]: 'string',
+  [FieldKey.DEVICE_BRAND]: 'string',
+  [FieldKey.DEVICE_LOCALE]: 'string',
+  [FieldKey.DEVICE_UUID]: 'string',
+  [FieldKey.DEVICE_ARCH]: 'string',
+  [FieldKey.DEVICE_BATTERY_LEVEL]: 'number',
+  [FieldKey.DEVICE_ORIENTATION]: 'string',
+  [FieldKey.DEVICE_SIMULATOR]: 'boolean',
+  [FieldKey.DEVICE_ONLINE]: 'boolean',
+  [FieldKey.DEVICE_CHARGING]: 'boolean',
+  [FieldKey.GEO_COUNTRY_CODE]: 'string',
+  [FieldKey.GEO_REGION]: 'string',
+  [FieldKey.GEO_CITY]: 'string',
+  [FieldKey.ERROR_TYPE]: 'string',
+  [FieldKey.ERROR_VALUE]: 'string',
+  [FieldKey.ERROR_MECHANISM]: 'string',
+  [FieldKey.ERROR_HANDLED]: 'boolean',
+  [FieldKey.STACK_ABS_PATH]: 'string',
+  [FieldKey.STACK_FILENAME]: 'string',
+  [FieldKey.STACK_PACKAGE]: 'string',
+  [FieldKey.STACK_MODULE]: 'string',
+  [FieldKey.STACK_FUNCTION]: 'string',
+  [FieldKey.STACK_IN_APP]: 'boolean',
+  [FieldKey.STACK_COLNO]: 'number',
+  [FieldKey.STACK_LINENO]: 'number',
+  [FieldKey.STACK_STACK_LEVEL]: 'number',
   // contexts.key and contexts.value omitted on purpose.
 
   // Transaction event fields.
-  'transaction.duration': 'duration',
-  'transaction.op': 'string',
-  'transaction.status': 'string',
+  [FieldKey.TRANSACTION_DURATION]: 'duration',
+  [FieldKey.TRANSACTION_OP]: 'string',
+  [FieldKey.TRANSACTION_STATUS]: 'string',
 
-  trace: 'string',
-  'trace.span': 'string',
-  'trace.parent_span': 'string',
+  [FieldKey.TRACE]: 'string',
+  [FieldKey.TRACE_SPAN]: 'string',
+  [FieldKey.TRACE_PARENT_SPAN]: 'string',
 
   // Field alises defined in src/sentry/api/event_search.py
-  project: 'string',
-  issue: 'string',
+  [FieldKey.PROJECT]: 'string',
+  [FieldKey.ISSUE]: 'string',
 } as const;
-assert(FIELDS as Readonly<{[key in keyof typeof FIELDS]: ColumnType}>);
+assert(FIELDS as Readonly<Record<FieldKey, ColumnType>>);
```

Comment:
Is this assert() still necessary or could we define the type when defining the variable? like `const FIELDS: Readonly<Record<FieldKey, ColumnType>>  = {...}`

Reply:
I was considering it, but wasn't sure if leaving it with `as const` was preferable as it gives better type hinting?

Reply:
nvm, checking shape will be a lot better for the mapping in the long wrong, thanks 👍